### ISSUE TITLE
[Bug] ListView, CollectionView - Scroll position is reset on tab change in android.

### ISSUE THREAD

**Issue by heebaek** [xamarin/Xamarin.Forms#8795]
i found this bug in my project using collectionview,  and i found this bug easily reproduce with default sample project using listview. 

### Description
ListView, CollectionView - Scroll position is reset on tab change 

### Steps to Reproduce

1. Create new project with default Shell Forms app. this project will contains default to-do like app.

2. add items enough until you can scroll down 

3. Scroll to bottom.

4. Go to about tab and go back to browse tab.

5. You can see Scroll position is reset in an android platform. 

### Expected Behavior
Scroll position not reset

### Actual Behavior
Scroll position reset

### Basic Information

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Screenshots

![Simulator Screen Shot - iPhone 11 - 2019-12-08 at 16 36 00](https://user-images.githubusercontent.com/12790588/70386128-ea22c900-19d8-11ea-92c2-5fc660a48e32.png)

**Comment by kingces95** [comment]
True. Likely by design but we'll let Ez tell us if that's really true.

**Comment by dmariogatto** [comment]
@kingces95 

Having a look at [ShellItemRendererBase](https://github.com/xamarin/Xamarin.Forms/blob/4.5.0/Xamarin.Forms.Platform.Android/Renderers/ShellItemRendererBase.cs), specifically `HandleFragmentUpdate`, the navigation source `ShellSectionChanged` results in the destruction of previous fragments. This causes the UI to be recreated when switching tabs, losing the previous state.

Is this by design?

I would've thought that it might be more efficient to show/hide fragments, and only inflate their views when navigation is first requested (similar to the `Push` navigation source)?

**Comment by krayknot** [comment]
I was not aware of this problem. I developed my app. You can get the sample by downloading krayknot - Android playstore.
Now, I will have to manage anyhow by some alternatives as this has disturbed my app's layout.

My app has 5 tabs and the 1st tab contains a Collectionview, the 5th tab has settings. Selecting Settings and then the Home(1st tab) resets the page.

Hope this will get resolved soon!

**Comment by codebreakerlegend2019** [comment]
I think our bug or problem have something in common. 
@dmariogatto do you have some workaround on this?

**Comment by krayknot** [comment]
> I think our bug or problem have something in common.
> @dmariogatto do you have some workaround on this?

in the first version, the tab with collectionview was on Index 0, now I have placed it on index 3. Index 0 has the search page now.

Just a workaround.

**Comment by codebreakerlegend2019** [comment]
> > I think our bug or problem have something in common.
> > @dmariogatto do you have some workaround on this?
> 
> in the first version, the tab with collectionview was on Index 0, now I have placed it on index 3. Index 0 has the search page now.
> 
> Just a workaround.

@krayknot do you have a video that it remove the lag?

**Comment by krayknot** [comment]
This is the app, I am working on:

https://play.google.com/store/apps/details?id=news.krayknot

At present the TAB sequence is Search, Categories, Home, Stories, Settings

Previously it was Home, Stories, Categories, Search and Settings.

So, whenever user changed any setting and comes back on "Home", all was reset. 

I tried to resolve it but gotcha know that this is a know bug and then I changed the sequence of the Tabs.

**Comment by codebreakerlegend2019** [comment]
> This is the app, I am working on:
> 
> https://play.google.com/store/apps/details?id=news.krayknot
> 
> At present the TAB sequence is Search, Categories, Home, Stories, Settings
> 
> Previously it was Home, Stories, Categories, Search and Settings.
> 
> So, whenever user changed any setting and comes back on "Home", all was reset. 
> 
> I tried to resolve it but gotcha know that this is a know bug and then I changed the sequence of the Tabs.
> 
> 

It seems fast and has no lag at all what Xamarin Forms Version you use??

**Comment by krayknot** [comment]
4.8

**Comment by codebreakerlegend2019** [comment]
> 4.8

Do you have Rest Api calls ?? From backend??

Where do you put the tab index thing?

**Comment by krayknot** [comment]
Yes there are REST Api calls. Its a tabbed page which is using Bottomnavigationview

**Comment by codebreakerlegend2019** [comment]
> Yes there are REST Api calls. Its a tabbed page which is using Bottomnavigationview

Wait wait, just to be clear. Have you used shell tab or the old tabbed page?

**Comment by Martin-4Spaces** [comment]
I have the same issue using Xamarin Forms Shell, Xamarin 4.8.0.1821, only Android.

**Comment by dmariogatto** [comment]
This is still an issue in the latest release (5.0.0.1874). UI state is lost when navigating between tabs. I've had to revert back to using a plain old TabbedPage to support older (slower) devices, otherwise, the UI redraw on switching tabs is really slow.

> Having a look at [ShellItemRendererBase](https://github.com/xamarin/Xamarin.Forms/blob/4.5.0/Xamarin.Forms.Platform.Android/Renderers/ShellItemRendererBase.cs), specifically `HandleFragmentUpdate`, the navigation source `ShellSectionChanged` results in the destruction of previous fragments. This causes the UI to be recreated when switching tabs, losing the previous state.

@hartez @rmarinho Would it be possible to confirm if what I've said in the above has any merit? If so, is this by design? Is there anything we can do to preserve UI state?

**Comment by malikberkane** [comment]
> This is still an issue in the latest release (5.0.0.1874). UI state is lost when navigating between tabs. I've had to revert back to using a plain old TabbedPage to support older (slower) devices, otherwise, the UI redraw on switching tabs is really slow.
> 
> > Having a look at [ShellItemRendererBase](https://github.com/xamarin/Xamarin.Forms/blob/4.5.0/Xamarin.Forms.Platform.Android/Renderers/ShellItemRendererBase.cs), specifically `HandleFragmentUpdate`, the navigation source `ShellSectionChanged` results in the destruction of previous fragments. This causes the UI to be recreated when switching tabs, losing the previous state.
> 
> @hartez @rmarinho Would it be possible to confirm if what I've said in the above has any merit? If so, is this by design? Is there anything we can do to preserve UI state?

Having the same issue, I have a tab with multiple custom controls nested, it literally takes ages to be redrawn every me I navigate to it, it also occurs when I set it as a flyout item.. This is just not acceptable.. any solution/ workaround to this problem?

**Comment by taublast** [comment]
The main issue here is a buggish behavour, one would say the UI state is reset, but in fact the renderers are recreated from scratch but forms controls state remains the same. Scroll to 100, switch tab, get back and you see the scrollview sitting at 0 while forms scrollview reports the property ScrollY being at 100. Please let's not talk about "by design" here..
The sad workaround is to save the scrollview position everytime the control scrolls, then on page onappearing invoke a ScrollTo  savedPosition on main thread.

**Comment by kasimier-vireq** [comment]
> The main issue here is a buggish behavour, one would say the UI state is reset, but in fact the renderers are recreated from scratch but forms controls state remains the same. Scroll to 100, switch tab, get back and you see the scrollview sitting at 0 while forms scrollview reports the property ScrollY being at 100. Please let's not talk about "by design" here.. The sad workaround is to save the scrollview position everytime the control scrolls, then on page onappearing invoke a ScrollTo savedPosition on main thread.

This seems also to be the case when using a CollectionView in a DataTemplate inside an other ListView/CollectionView.
Even with ListView RetailElement, the CollectionViews in the data templates become out-of-sync.
We have added ScrollPosition bindable property to the inner CollectionViews and are listening to changes of that property in order to call ScrollTo.
This works fine whenever we change ScrollPosition.
If ScrollPosition does not change, then we have no way of updating the CollectionView's actual scroll position, which is always reset when those CollectionViews become visible again.

**Comment by rudyspano** [comment]
I confirm this issue.
On any Shell based app, just add the following code to a page:
```
        protected override void OnPropertyChanged([CallerMemberName] string propertyName = null)
        {
            base.OnPropertyChanged(propertyName);

            if (propertyName == "Renderer")
            {

            }
        }
```
By putting a breakpoint and using sourcelink feature ,  you will see easily that the renderer of a page (and all child controls) is 
1) set to null and native views are disposed
2) set to a new renderer instance on tab new selection

I can understand the need to dispose native views (fragments) for shell items but I think Tabs should be an exception in order to have a smooth experience similarly to TabbedPage. The delay is only 300->500ms but that's hard to justify after a dev effort to migrate from TabbedPage to Shell...  

I tried to "patch" this behavior on native side but a lot of internal/private code is involved...
I also tried to preload, keep native views renderers/instances as a dirty workaround...
Any workaround on your side ?

**Comment by FloexTek** [comment]
Is this ever gonna get fixed?

**Comment by taublast** [comment]
@rudyspano sorry i missed that message before. Another workaround is to:
1. Use instance instead of datatemplate, istead of `<shellitem datatemplate=blabla/>` use directly `<MyAwesomeTab/>`
2. Use lazy loading of tab content if you think it's laggy. So at launch time you'd have an empty tab content, and when it is shown at the first time (very quickly as you might guess) set with delay on ui thread the real content, even with fade-in effect.
3. You can switch to using Sharpnado Tabs, along with fully alternative way of having tabs it has lazy loading support. In that case you would loose support from xamarin shell, but you can have your own shell implementation, can help you in that matter if needed.
4. Can think about migrating to maui as let's be honest xamarin issues are not a priority any more.
